### PR TITLE
198353 - Extension de la fonction d'export aux ilots et baies

### PR DESCRIPTION
**Pull request by francoisedumas**
[User story](https://app.activecollab.com/170818/projects/278/tasks/3309)

Add export for islets
Add export for bays
Scope call for ports export

### CODE REVIEW

File: app/controllers/bays_controller.rb [hunk 1]
```diff
@@ -37,7 +39,7 @@ def show
                pdf: 'frame',
                zoom: 0.75
       end
-      format.txt { send_data Frame.to_txt(@servers_per_frames, params[:bg]) }
+      format.txt { send_data Frame.to_txt(@servers_per_frames[@bay.islet.room_id], params[:bg]) }
```
Comment (nicolas-brousse) [inline app/controllers/bays_controller.rb:42]: Not sure of the needs of this changes

File: app/controllers/ports_controller.rb [hunk 2]
```diff
@@ -9,6 +9,8 @@ def index
       @frames = @room.frames
       if params[:islet].present?
         @frames = @frames.joins(:bay => :islet).where('islets.name = ?', params[:islet])
+      elsif params[:bay_id].present?
+        @frames = @frames.joins(:bay).where('bays.id = ?', params[:bay_id])
```
Comment (nicolas-brousse) [inline app/controllers/ports_controller.rb:13]: What this used for?

File: app/controllers/islets_controller.rb [hunk 3]
```diff
@@ -1,10 +1,42 @@
...
+
+    respond_to do |format|
+      format.pdf do
+        render layout: 'pdf.html',
+               template: "rooms/show.pdf.erb",
+               show_as_html: params[:debug].present?,
+               pdf: 'frame',
+               zoom: 0.75
+      end
+      format.txt { send_data Frame.to_txt(@servers_per_frames[@room.id], params[:bg]) }
```
[the remark this answers is not in the document]
Comment (francoisedumas) [reply app/controllers/islets_controller.rb:36]: As I am re-using the view `"rooms/show.pdf.erb"` I kept the structure of the hash sent to the view and there is a room level. It enables the display of the room the islet is in
Screenshot 2566-01-18 at 18 06 29

File: app/controllers/bays_controller.rb (hunk 1 above)
Comment (francoisedumas) [reply app/controllers/bays_controller.rb:42]: As I am re-using the view "rooms/show.pdf.erb" I kept the structure of the hash sent to the view and there is a room level. It enables the display of the room the bay is in.
Screenshot 2566-01-18 at 18 06 29

**Pull request merged**